### PR TITLE
Python 3.10 support (and migrate to GitHub actions)

### PR DESCRIPTION
**Pull request by stefanor**
`Iterable` moved to `collections.abc` in `3.3`

Fixes: #195

While we're here, migrate to GitHub actions, so we can have CI, again.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (fabaff) [inline .github/workflows/ci.yml:9]: Everything below Python 3.7 is EOL. I suggest to stick to supported Python releases.

Comment (stefanor) [reply .github/workflows/ci.yml:9]: Yeah, that'd be sensible. I don't know what the maintainers want to be supporting, though?

**Comment by shimamizu** [comment]
@dlyssenko this seems pretty straight forward? Would be good to get 3.10 support.

**Comment by dlyssenko** [comment]
@stefanor, @shimamizu, @fabaff. W.r.t. switching over to Github workflow ([ebdbb24](https://github.com/arista-eosplus/pyeapi/pull/223/commits/ebdbb2486e2656ce6d691d0ef15c7cd7f4a8c0e2)) - I'm myself in favor of this decision, probably will approve after checking with my team.

For the supported versions in CI ([cc9c584](https://github.com/arista-eosplus/pyeapi/pull/223/commits/cc9c584e4a3167e3c1624cccb6bc0d9c9bcdbc1c)) - I'd think we still need even 2.7 - some of our customers might still use it, documentations is still being built using sphinx on python2, EOS integration still relies on python2 and system tests are not yet entirely migrated over to python3. We' making an effort to migrate completely to python3, but it takes time, until then we'd need to include 2.7 into the CI workflow.

for the same reason I cannot approve [81754f5](https://github.com/arista-eosplus/pyeapi/pull/223/commits/81754f57eb095703cc474f527a0915360af76f68) - but PR #235, #239 ensure compatibility, so that one could be dropped entirely.

**Review by dlyssenko — APPROVED** [review]

**Comment by dlyssenko** [comment]
1. I've merged [ebdbb24](https://github.com/arista-eosplus/pyeapi/pull/223/commits/ebdbb2486e2656ce6d691d0ef15c7cd7f4a8c0e2) manually through `git cherry-pick`, 
2. will also update `python-version` manually to extend to 2.7
3. the other commits  (`cc9c584`, `81754f5`) are not required, as per my update earlier.

closing this PR with that.

**Pull request closed without merging**
